File: rust/skialin-core/src/color.rs
pub type Color = u32;

pub const fn argb(a: u8, r: u8, g: u8, b: u8) -> Color {
    ((a as u32) << 24) | ((r as u32) << 16) | ((g as u32) << 8) | (b as u32)
}

pub const fn rgb(r: u8, g: u8, b: u8) -> Color {
    argb(0xff, r, g, b)
}

pub const BLACK: Color = argb(0xff, 0, 0, 0);
pub const WHITE: Color = argb(0xff, 0xff, 0xff, 0xff);
pub const TRANSPARENT: Color = argb(0, 0, 0, 0);
pub const RED: Color = argb(0xff, 0xff, 0, 0);
pub const GREEN: Color = argb(0xff, 0, 0xff, 0);
pub const BLUE: Color = argb(0xff, 0, 0, 0xff);

File: src/main/kotlin/org/skialin/Color.kt
package org.skialin

typealias Color = Int

object Colors {
    fun argb(
        a: Int,
        r: Int,
        g: Int,
        b: Int,
    ): Color = (a shl 24) or (r shl 16) or (g shl 8) or b

    fun rgb(
        r: Int,
        g: Int,
        b: Int,
    ): Color = argb(0xff, r, g, b)

    val BLACK: Color = rgb(0, 0, 0)
    val WHITE: Color = rgb(0xff, 0xff, 0xff)
    val TRANSPARENT: Color = argb(0, 0, 0, 0)
    val RED: Color = rgb(0xff, 0, 0)
    val GREEN: Color = rgb(0, 0xff, 0)
    val BLUE: Color = rgb(0, 0, 0xff)
}
